### PR TITLE
Remove MS SQL JDBC driver from the Keycloak product

### LINKED ISSUE #22983: Remove MS SQL JDBC driver from the Keycloak product
### Description

Similar to #22598, remote the MSSQL driver. 

As a first step, the driver needs to be removed. Documentation will be handled in a second step.

### PR DESCRIPTION
**Pull request by ahus1**
Closes #22983

### CODE REVIEW

**Review by mhajas — COMMENTED** [review]
Should we add a note also to the Release notes? https://github.com/keycloak/keycloak/blob/main/docs/documentation/release_notes/topics/23_0_0.adoc

**Comment by ahus1** [comment]
Thank you @mhajas for catching this, updated!

**Review by mhajas — DISMISSED** [review]
Thank you @ahus1!

**Review by vmuzikar — CHANGES_REQUESTED** [review]
Could we perhaps use similar approach as [here](https://github.com/keycloak/keycloak/commit/aae4569071fdb115394dfb5dce506a57852d1706#diff-8c389ae6ee129b90bd762d101c2c3c00fe0b0af41cb3b352150706552682c28b)? I.e. a specific maven profile to remove it and the same profile to add it to the testsuite?

**Review by vmuzikar — COMMENTED** [review]
The driver should be removed from product only, in community it should remain.

**Comment by mhajas** [comment]
> The driver should be removed from product only, in community it should remain.

If we want to follow the same as for the Oracle driver, this should not be done in this PR. See your comment in the Oracle PR: https://github.com/keycloak/keycloak/pull/22453#pullrequestreview-1586740986

**Comment by mhajas** [comment]
> Could we perhaps use similar approach as [here](https://github.com/keycloak/keycloak/commit/aae4569071fdb115394dfb5dce506a57852d1706#diff-8c389ae6ee129b90bd762d101c2c3c00fe0b0af41cb3b352150706552682c28b)? I.e. a specific maven profile to remove it and the same profile to add it to the testsuite?

The same as above, we don't want to do this here if we follow the same as for the Oracle

**Comment by vmuzikar** [comment]
@mhajas After discussion with @stianst, we decided to take a bit different approach and include all drivers in upstream, see also https://github.com/keycloak/keycloak/issues/22999

**Comment by ahus1** [comment]
Will wait for #23006 to be merged first, then this will be rebased/updated to be simpler to be merged to the 22 branch.

**Comment by ahus1** [comment]
@vmuzikar - this now applies the changes to similar to the Oracle driver. I also verified that it can be cherry-picked without modifications to KC22. 

Please review 

Thanks!

/cc @mabartos

**Review by mabartos — APPROVED** [review]
@ahus1 LGTM. Works as expected.

**Comment by ahus1** [comment]
Hi @vmuzikar - if you agree, this should now be ready to be merged. Thanks!

File: quarkus/tests/junit5/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java [hunk 1]
```diff
@@ -389,6 +389,10 @@ private Path prepareDistribution() {
             if (!inited || (reCreate || !dPath.toFile().exists())) {
                 FileUtil.deleteDirectory(dPath);
                 ZipUtils.unzip(distFile.toPath(), distRootPath);
+                if (System.getProperty("product") != null) {
+                    // MS SQL Server driver might be excluded if running as a product build
+                    copyProvider(dPath, "com.microsoft.sqlserver", "mssql-jdbc");
+                }
```
Comment (vmuzikar) [inline quarkus/tests/junit5/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java:395]: I think we don't necessarily need to add the driver for all tests but maybe rather tie it to the `@WithDatabase` annotation. But it's not a blocker for me, feel free to merge without it.

**Review by vmuzikar — APPROVED** [review]
@ahus1 Thank you, LGTM.

File: quarkus/tests/junit5/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java (hunk 1 above)
Comment (vmuzikar) [reply quarkus/tests/junit5/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java:395]: Created similar issue to fix Oracle as well: https://github.com/keycloak/keycloak/issues/23058
CC: @mabartos

Comment (ahus1) [reply quarkus/tests/junit5/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java:395]: Thank you @vmuzikar - I'd like to continue without changing this further at least not for KC 22.0.2. Thanks!

**Pull request merged**